mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -61,20 +67,70 @@ class LogisticRegressionModel (
 
   override protected def predictPoint(dataMatrix: Vector, weightMatrix: Vector,
```

Comment:
I know this was in the existing code, but `dataMatrix` isn't conceptually a matrix. `weightMatrix` now is, but it's represented as a `Vector`. There is a `Matrix` class that's more suitable, it seems. I know, changing that would change the API. There could be a second method accepting `Vector` and `Matrix` though. The existing one is for binary, new one for multinomial, and the former can call the latter, even.